File: test/table-base64/go/main.go
package main

import (
	b "base64test"
	"bufio"
	"encoding/hex"
	"fmt"
	"os"
)

func packed(data []byte) string {
	if len(data) == 0 {
		return "-"
	}
	return hex.EncodeToString(data)
}

func main() {
	s := bufio.NewScanner(os.Stdin)
	s.Buffer(make([]byte, 65536), 1024*1024)
	for s.Scan() {
		text, err := hex.DecodeString(s.Text())
		if err != nil {
			panic(err)
		}
		var v b.Blob
		var r b.TableReport
		b.BlobFromJson(&v, text, &r)
		if r.Malformed {
			fmt.Println("1 0 0 - -")
			continue
		}
		out := make([]byte, b.BlobToJsonMeasure(&v))
		if b.BlobToJson(&v, out) != int64(len(out)) {
			panic("writer size")
		}
		fmt.Printf("0 %d %d %s %s\n", r.Clamped, r.KindMismatch, packed(v.Payload[:v.PayloadLength]), packed(out))
	}
	if err := s.Err(); err != nil {
		panic(err)
	}
}

File: test/table-base64/rust/main.rs
use base64test::{Blob, TableReport, blob_from_json, blob_to_json, blob_to_json_measure};
use std::io::{self, BufRead};

fn hex(bytes: &[u8]) -> String {
    if bytes.is_empty() {
        return "-".into();
    }
    bytes.iter().map(|b| format!("{b:02x}")).collect()
}

fn main() {
    for line in io::stdin().lock().lines() {
        let line = line.expect("input");
        let text: Vec<u8> = (0..line.len())
            .step_by(2)
            .map(|i| u8::from_str_radix(&line[i..i + 2], 16).expect("hex"))
            .collect();
        let mut value = Blob::default();
        let mut report = TableReport::default();
        blob_from_json(&mut value, &text, &mut report);
        if report.malformed {
            println!("1 0 0 - -");
            continue;
        }
        let mut out = vec![0; blob_to_json_measure(&value) as usize];
        assert_eq!(blob_to_json(&value, &mut out), out.len() as i64);
        println!(
            "0 {} {} {} {}",
            report.clamped,
            report.kind_mismatch,
            hex(&value.payload[..value.payload_length as usize]),
            hex(&out)
        );
    }
}
